contrib/userscript/unnoticed.user.js

```diff
@@ -0,0 +1,201 @@
+        var p300 = c300 / ctotal;
+        if(c100 == 0 && c50 == 0 && cmiss == 0){
+            if(mods.includes("HD"))
+                return "XH";
+            else
+                return "X";
+        }else if(p300 > 0.9 && c50 / ctotal < 0.01 && cmiss == 0){
+            if(mods.includes("HD"))
+                return "SH";
+            else
+                return "S";
+        }else if(p300 > 0.8 && cmiss == 0 || p300 > 0.9){
+            return "A";
+        }else if(p300 > 0.7 && cmiss == 0 || p300 > 0.8){
+            return "B";
+        }else if(c300 > 0.6){
+            return "C";
+        }else{
+            return "D";
+        }
+    }
+
+    function mods(enabled_mods){
+        var return_array = [];
+        for(var mod in mods_enum){
+            if((mods_enum[mod] & enabled_mods) != 0)
+                return_array.push(mod);
+        }
+        return return_array;
+    }
+
+    function mods_string(enabled_mods){
+        if(mods(enabled_mods).length > 0)
+            return  mods(enabled_mods)
+                    .join(",")
+                    .replace("DT,NC", "NC")
+                    .replace("SD,PF", "PF");
+        else
+            return "None";
+    }
+
+    var api_base = "https://p9bztcmks6.execute-api.us-east-1.amazonaws.com/unnoticed/proxy?b=";
+
+    // check if a on an unranked beatmap page
+    if($(".beatmapTab").length > 0 && $(".scoreLeader").length === 0){
+        console.log("beatmap unranked, retrieving scores via unnoticed api");
+        var beatmap_id = $(".beatmapTab.active")
+                        .attr("href").split("/").pop().split("&")[0];
+        console.log(beatmap_id);
+        GM_xmlhttpRequest({
+            method: "GET",
+            url: api_base + beatmap_id,
+            onload: function(response_raw){
+                var response = JSON.parse(response_raw.responseText);
+
+                var scores = [];
+
+                response.scores[beatmap_id].forEach(function(score){
+                    var exists = false;
+
+                    scores.forEach(function(score_check, index){
+                        if(score_check.player_id == score.player_id){
+                            exists = true;
+                            if(score_check.score < score.score)
+                                scores[index] = score;
+                        }
+                    });
+
+                    if(!exists){
+                        scores.push(score);
+                    }
+                });
+
+                scores = scores.sort(function(a, b){ return a.score - b.score; })
+                               .reverse();
+
+                if(scores.length > 0){
+                    $('<div style="margin:20px auto; background: rgb(208, 231, 249); border-radius: 5px; width: 50%; padding: 15px;">'
+                    + '<center>This map is UNRANKED.<br />'
+                    + 'As such, it doesn\'t reward any pp and scores are retrieved via Unnoticed.</center>'
```

Comment:
Can we change to "[Unnoticed]" and add a link to this repo?

Reply:
yes